jupyterlab_gcedetails/src/components/resource_utilization_charts.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import { ServerConnection } from '@jupyterlab/services';
+import { AreaChartWrapper } from './chart_wrapper';
+import {
+  Utilization,
+  AREA_CHART_BLUE,
+  AREA_CHART_ORANGE,
+  UTILIZATION_CHART_PROPERTIES,
+  STYLES,
+} from '../data';
+
+interface State {
+  data?: Utilization[];
+  receivedError: boolean;
+}
+
+export class ResourceUtilizationCharts extends React.Component<{}, State> {
+  private readonly serverSettings = ServerConnection.defaultSettings;
```

Comment:
Not sure if it's overengineering but maybe we can have a server object initialized in a parent component and have that be passed down as a props instead of duplicating this code.

Reply:
Yeah I would suggest factoring out the code that interacts with the server-side API to a common module within the extension so it can be shared by both this component and the parent `VmDetails`.